test/iht/views/application/ShareableElementInputViewBehaviour.scala

```diff
@@ -87,7 +88,11 @@ trait ShareableElementInputViewBehaviour[A] extends ViewTestHelper {
     "show a return link" in {
       val link = doc.getElementById("return-button")
       link.text shouldBe returnLinkText
-      link.attr("href") shouldBe returnLinkUrl
+      if(linkHash > ""){
```

Comment:
Better, clearer, to use .nonEmpty here. Personally I'd probably write it like this:
val expectedResult = if(linkHash.nonEmpty) attrib.target.url + linkHash else attrib.target.url
cancelButton.attr("href") shouldBe expectedResult
but that's perhaps more a matter of personal taste.
Also, this bit of code is repeated 4 times throughout tests, so I'd put it somewhere common such as TestHelper.